src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -61,14 +82,36 @@ export function datatableToCSV(
   }
 
   // Convert the array of row objects to an array of row arrays
-  const csvRows = rows.map((row) => {
-    return sortedColumnIds.map((id) =>
-      escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
-    );
-  });
+  const csvRows = rows
+    .map((row) => {
+      return sortedColumnIds.map((id) =>
+        escapeValues(raw ? row[id] : formatters[id].convert(row[id]))
+      );
+    })
+    .sort(rowSortPredicate(sortedColumnIds, columnSorting));
 
   return (
     [header, ...csvRows].map((row) => row.join(csvSeparator)).join(LINE_FEED_CHARACTER) +
     LINE_FEED_CHARACTER
   ); // Add \r\n after last line
 }
+
+function rowSortPredicate(
+  sortedColumnIds: string[],
+  columnSorting?: EuiDataGridColumnSortingConfig[]
+) {
+  if (!columnSorting) return () => 0;
+
+  const columnIdMap = new Map(columnSorting.map(({ id }) => [id, sortedColumnIds.indexOf(id)]));
+  return (rowA: string[], rowB: string[]) => {
+    return columnSorting.reduce((acc, { id, direction }) => {
+      const i = columnIdMap.get(id) ?? -1;
+      if (i < 0) return acc;
+
+      const a = rowA[i];
+      const b = rowB[i];
+      const emptyValueSort = a === '' ? 1 : b === '' ? -1 : 0; // always put empty values at bottom
+      return acc || emptyValueSort || a.localeCompare(b) * (direction === 'asc' ? 1 : -1);
+    }, 0);
+  };
+}
```

Comment:
Row sorting to match visible sorting order.

<img width="806" alt="image" src="https://github.com/user-attachments/assets/49ee8520-caad-446d-a5e1-efa629e912d1">

<img width="954" alt="image" src="https://github.com/user-attachments/assets/0f862f78-4734-4b6b-a8f2-637e2a43e05c">

Reply:
This is now handled in `vis. getExportDatatables`